src/main/java/tc/oc/pgm/rotation/RotationManager.java

```diff
@@ -0,0 +1,220 @@
+  private boolean isEvaluatingPlayerCount = true;
+  /** When a {@link PGMMap} is manually set next, it overrides the rotation order * */
+  private PGMMap overriderMap;
+
+  public RotationManager(MatchManager matchManager, Logger logger, File rotationsFile) {
+    this.matchManager = matchManager;
+    this.rotationsFile = rotationsFile;
+
+    if (!rotationsFile.exists()) {
+      try {
+        FileUtils.copyInputStreamToFile(
+            PGM.GLOBAL.get().getResource("rotations.yml"), rotationsFile);
+      } catch (IOException e) {
+        logger.log(Level.SEVERE, "Failed to create the rotations.yml file", e);
+      }
+    }
+
+    loadRotations();
+  }
+
+  private void loadRotations() {
+    FileConfiguration rotationsFileConfiguration =
+        YamlConfiguration.loadConfiguration(rotationsFile);
+
+    rotationsFileConfiguration
+        .getKeys(false)
+        .forEach(
+            key ->
+                rotations.add(
+                    new Rotation(rotationsFileConfiguration.getConfigurationSection(key), key)));
+    rotations.forEach(Rotation::load);
+
+    rotations.forEach(
+        rotation -> {
+          if (!rotation.isEnabled()) rotations.remove(rotation);
+        });
+
+    setActiveRotation(getRotationByName("default"));
+  }
+
+  private void setActiveRotation(Rotation activeRotation) {
+    this.activeRotation = activeRotation;
+  }
+
+  public Rotation getActiveRotation() {
+    return activeRotation;
+  }
+
+  void setEvaluatingPlayerCount(boolean evaluatingPlayerCount) {
+    this.isEvaluatingPlayerCount = evaluatingPlayerCount;
+  }
+
+  public boolean isEvaluatingPlayerCount() {
+    return isEvaluatingPlayerCount;
+  }
+
+  public List<Rotation> getRotations() {
+    return rotations;
+  }
+
+  private void saveCurrentPosition() {
+    FileConfiguration rotationFileConfiguration =
+        YamlConfiguration.loadConfiguration(rotationsFile);
+    rotationFileConfiguration.set(
+        activeRotation.getName() + ".position", activeRotation.getPosition());
+
+    try {
+      rotationFileConfiguration.save(rotationsFile);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void recalculateActiveRotation() {
+    int activePlayers = getActivePlayers();
+    int[] playerCounts = new int[rotations.size()];
+
+    AtomicInteger count = new AtomicInteger(0);
+    rotations.forEach(
+        rotation -> {
+          if (rotation.isEnabled()) {
+            playerCounts[count.get()] = rotation.getPlayers();
+            count.getAndIncrement();
+          }
+        });
+
+    int distance = Math.abs(playerCounts[0] - activePlayers);
+    int accuratePlayerCountIndex = 0;
+
+    for (int i = 1; i < playerCounts.length; i++) {
+      int idistance = Math.abs(playerCounts[i] - activePlayers);
+      if (idistance < distance) {
+        accuratePlayerCountIndex = i;
+        distance = idistance;
+      }
+    }
+
+    updateRotation(playerCounts, accuratePlayerCountIndex);
+  }
```

Comment:
This code is very cryptic, and hard to understand exactly what's going on. What does this do exactly?

Reply:
Made it less cryptic I guess, https://github.com/Electroid/PGM/pull/114/commits/223f8e1eaea620bb129a78e3b6c0eb629a5bc193